ql/src/java/org/apache/hadoop/hive/ql/metadata/PartitionIterable.java

```diff
@@ -99,7 +99,16 @@ private void getNextBatch() {
           batchCounter++;
         }
         try {
-          if (isAuthRequired) {
+          if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
+            List<Partition> partitions = new ArrayList<>();
```

Comment:
why do we need new List var?

Reply:
I removed this change, it is not needed anymore.